swarm/storage/netstore.go

```diff
@@ -52,6 +52,8 @@ type NetStore struct {
 	closeC            chan struct{}
 }
 
+var fetcherTimeout = 2 * time.Minute // timeout to cancel the fetcher even if requests are coming in
```

Comment:
do we really need this?

Reply:
This is an additional safety cancelation, beside the hop counter limit. It is an internal cancelation, as hop counts are submitted from outside and can be arbitrary counts received.

Reply:
understood